javatests/arcs/android/storage/service/BindingContextTest.kt

```diff
@@ -121,6 +127,24 @@ class BindingContextTest {
     assertThat(operations.operations).isEqualTo(message.operations)
   }
 
+  @Test
+  fun registerCallback_errorDuringRegistration_propagatesException(): Unit = runBlocking {
+    val fakeStore = FakeActiveStore<CrdtData, CrdtOperation, Int>(
+      StoreOptions(storageKey, CountType())
+    )
+
+    val bindingContext = buildContext(storeProvider = { fakeStore })
+    val callback = DeferredProxyCallback()
+
+    val e = assertFailsWith(CrdtException::class) {
+      suspendForRegistrationCallback {
+        bindingContext.registerCallback(callback, it)
+      }
+    }
+    assertThat(e).hasCauseThat().hasCauseThat()
+      .hasMessageThat().contains("Intentionally thrown!")
```

Comment:
This relates to my comment in `FakeActiveStore.kt`: it's not obvious reading this test case where the `"Intentionally thrown!"` error message is coming from. Intentionally thrown from where? You have to know to go digging into the implementation of `FakeActiveStore` to find it. Having to go digging anywhere is not ideal when trying to understand a test case, but particularly not so when the place you need to look isn't the test file nor the class under test. 

This is why I'd suggest to override the `on` method directly in the test case. Then you'll be able to very clearly see where it came from:

```
val fakeStore = object : NoopActiveStore {
  override fun on() = throw BlahBlah()
}

...

assertThat(e).hasCauseThat().isInstanceOf(BlahBlah)
```